utils/build/docker/golang/parametric/datadog.go

```diff
@@ -42,8 +42,13 @@ func (s *apmClientServer) startSpanHandler(w http.ResponseWriter, r *http.Reques
 func (s *apmClientServer) StartSpan(ctx context.Context, args *StartSpanArgs) (ddtrace.Span, error) {
 	var opts []tracer.StartSpanOption
 	if p := args.ParentId; p > 0 {
-		parent := s.spans[p]
-		opts = append(opts, tracer.ChildOf(parent.Context()))
+		if span, ok := s.spans[p]; ok {
+			opts = append(opts, tracer.ChildOf(span.Context()))
+		} else if spanContext, ok := s.spanContexts[p]; ok {
+			opts = append(opts, tracer.ChildOf(spanContext))
+		} else {
+			return nil, fmt.Errorf("parent span not found")
+		}
```

Comment:
Yup since this functionality isn't merged we can do this in a follow up PR. 

Side note: Would it make sense to test this functionality via a weblog endpoint? Weblog endpoints have ddtrace integrations enabled and those integrations extract distributed tracing headers. If we use weblog we can validate whether integrations support generating a span link for conflicting headers. I think this could be a better test.

Reply:
Yes I agree. We can migrate these tests to weblog. 👍

Reply:
yes PLEASE!